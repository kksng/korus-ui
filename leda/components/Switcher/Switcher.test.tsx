import React from 'react';
import {
  render,
  fireEvent,
} from '@testing-library/react';

import { Switcher } from './index';

const validName = 'switcher';
const on = true;
const off = false;

describe('Switcher spanpshot collection', () => {
  test('is Switcher render right?', () => {
    const { container, getByText } = render(<Switcher>Switch</Switcher>);
    const wrapper = container.querySelectorAll('.switcher-wrapper')[0];
    const handler = container.querySelectorAll('.switcher-handle')[0];
    const label = container.querySelectorAll('.switcher-label')[0];

    expect(wrapper)
      .toBeInTheDocument();

    expect(handler)
      .toBeInTheDocument();

    expect(label)
      .toBeInTheDocument();

    expect(getByText('Switch'))
      .toBeInTheDocument();

    expect(container.firstChild)
      .toMatchSnapshot();
  });
  test('is Switcher render right if value set True and False?', () => {
    const { container, rerender } = render(<Switcher value={on}>Switch</Switcher>);
    const wrapper = container.querySelectorAll('.switcher-wrapper')[0];

    expect(wrapper)
      .toBeInTheDocument();

    expect(wrapper)
      .toHaveClass('active');

    expect(container.firstChild)
      .toMatchSnapshot();

    rerender(<Switcher value={off}>Switch</Switcher>);

    expect(wrapper)
      .not
      .toHaveClass('active');

    expect(container.firstChild)
      .toMatchSnapshot();
  });
});
describe('Switcher attributes test collection', () => {
  test('is Switcher render right with isDisabled attibutes?', () => {
    const { container } = render(<Switcher isDisabled>Switch</Switcher>);
    const wrapper = container.querySelectorAll('.switcher-wrapper')[0];

    expect(wrapper)
      .toHaveClass('disabled');
  });
  test('is Switcher render right with children elements?', () => {
    const { container } = render(<Switcher><div className="lvl1"><span className="lvl2">Switch</span></div></Switcher>);
    const childrenA = container.querySelectorAll('div.lvl1');
    const childrenB = container.querySelectorAll('span.lvl2');

    expect(childrenA)
      .toHaveLength(1);

    expect(childrenB)
      .toHaveLength(1);
  });
});
describe('Switcher event handler collection', () => {
  test('is Switcher worj right with onChange event handler?', () => {
    const onChange = jest.fn();
    const { container } = render(<Switcher name={validName} value={on} onChange={onChange}>Switcher</Switcher>);
    const wrapper = container.querySelectorAll('.switcher-wrapper')[0];
    const label = container.querySelectorAll('.switcher-label')[0];

    expect(label)
      .not
      .toHaveClass('active');

    fireEvent.click(wrapper);

    expect(onChange)
      .lastCalledWith(expect.objectContaining({
        component: expect.objectContaining({
          name: validName,
          value: off,
        }),
      }));
  });
});
