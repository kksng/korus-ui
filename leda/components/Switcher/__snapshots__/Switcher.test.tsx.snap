// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher spanpshot collection is Switcher render right if value set True and False? 1`] = `
<div>
  <div
    class="switcher-wrapper active"
  >
    <div
      class="switcher-handle"
    />
  </div>
  <span
    class="switcher-label"
  >
    Switch
  </span>
</div>
`;

exports[`Switcher spanpshot collection is Switcher render right if value set True and False? 2`] = `
<div>
  <div
    class="switcher-wrapper"
  >
    <div
      class="switcher-handle"
    />
  </div>
  <span
    class="switcher-label"
  >
    Switch
  </span>
</div>
`;

exports[`Switcher spanpshot collection is Switcher render right? 1`] = `
<div>
  <div
    class="switcher-wrapper"
  >
    <div
      class="switcher-handle"
    />
  </div>
  <span
    class="switcher-label"
  >
    Switch
  </span>
</div>
`;
